File: backend/test_watchlist_endpoint.py
#!/usr/bin/env python3
"""
Test the watchlist endpoint
"""
import requests

def test_watchlist_endpoint():
    """Test the /api/watchlist endpoint"""
    print("=== Testing Watchlist Endpoint ===\n")
    
    try:
        # Test the watchlist endpoint
        url = "http://127.0.0.1:8000/api/watchlist"
        print(f"Making request to: {url}")
        
        response = requests.get(url, timeout=15)
        print(f"Status Code: {response.status_code}")
        
        if response.status_code == 200:
            data = response.json()
            print("✅ Watchlist endpoint working!")
            print(f"Number of items: {len(data.get('items', []))}")
            for item in data.get('items', [])[:3]:  # Show first 3 items
                print(f"  - {item.get('ticker', 'N/A')}: {item.get('company_name', 'N/A')}")
        else:
            print(f"❌ Error: {response.status_code}")
            print(f"Response: {response.text}")
            
    except requests.exceptions.ConnectionError:
        print("❌ Connection failed")
    except requests.exceptions.Timeout:
        print("❌ Request timed out")
    except Exception as e:
        print(f"❌ Error: {e}")

if __name__ == "__main__":
    test_watchlist_endpoint()

File: test-watchlist-endpoint.js
const https = require('https');

// Test the watchlist endpoint directly
const API_BASE_URL = 'https://dx0w31lbc1.execute-api.eu-west-1.amazonaws.com/production';

function testEndpoint(path, description) {
    return new Promise((resolve, reject) => {
        console.log(`\n🧪 Testing ${description}...`);
        console.log(`URL: ${API_BASE_URL}${path}`);
        
        const options = {
            hostname: 'dx0w31lbc1.execute-api.eu-west-1.amazonaws.com',
            port: 443,
            path: `/production${path}`,
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        };

        const req = https.request(options, (res) => {
            let data = '';
            
            console.log(`Status: ${res.statusCode}`);
            console.log(`Headers:`, res.headers);
            
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                try {
                    const jsonData = JSON.parse(data);
                    console.log(`✅ ${description} - Success`);
                    console.log('Response:', JSON.stringify(jsonData, null, 2));
                    resolve({ status: res.statusCode, data: jsonData });
                } catch (e) {
                    console.log(`Response (raw):`, data);
                    resolve({ status: res.statusCode, data: data });
                }
            });
        });

        req.on('error', (error) => {
            console.log(`❌ ${description} - Error:`, error.message);
            reject(error);
        });

        req.setTimeout(10000, () => {
            console.log(`⏰ ${description} - Timeout`);
            req.destroy();
            reject(new Error('Request timeout'));
        });

        req.end();
    });
}

async function runTests() {
    console.log('🚀 Testing Stock Analysis API Endpoints');
    console.log('==========================================');
    
    try {
        // Test health endpoint first
        await testEndpoint('/health', 'Health Check');
        
        // Test root API endpoint
        await testEndpoint('/api', 'API Root');
        
        // Test watchlist endpoint (the problematic one)
        await testEndpoint('/api/watchlist', 'Watchlist Endpoint');
        
        // Test other endpoints to see what's working
        await testEndpoint('/api/search', 'Search Endpoint');
        
        console.log('\n✅ All tests completed');
        
    } catch (error) {
        console.log('\n❌ Test suite failed:', error.message);
    }
}

runTests();